assets/js/components/ReportError.js

```diff
@@ -68,12 +69,16 @@ export default function ReportError( { moduleSlug, error } ) {
 			err.selectorData.name === 'getReport'
 	);
 
+	const determineError = memize( ( errorsA, errorsB ) => {
+		return {
+			...( errorsA.length ? errorsA[ 0 ] : errorsB[ 0 ] ),
+		};
+	} );
+
 	const showRetry = !! retryableErrors.length;
 
 	const errorTroubleshootingLinkURL = useSelect( ( select ) => {
-		const err = {
-			...( showRetry ? retryableErrors[ 0 ] : errors[ 0 ] ),
-		};
+		const err = determineError( retryableErrors, errors );
```

Comment:
If the concern here is to avoid unnecessary re-renders of the component, this shouldn't be necessary because `useSelect` only triggers a re-render when the output of the function is different. In this case, the output is a string, which wouldn't change based on the identity of the object.

It's also worth noting that `useSelect` uses a shallow equality comparison for this so even if a new object was returned from the `useSelect` every time, it wouldn't force a re-render on every invocation. See https://github.com/WordPress/gutenberg/blob/3e867dd7c6560e4b2cb9a59cc02856e055be6142/packages/data/src/components/use-select/index.js#L126-L138

The other concern here would be that a new `err` object passed to the `getErrorTroubleshootingLinkURL` selector below would invoke its resolver every time due to a "new" set of arguments it's called with. Internally it uses https://github.com/aduth/equivalent-key-map which uses deep equality but is loose in terms of the identity comparison, so this would also be okay!

The problem with new objects is more of a risk when used in other places which are more strict about the comparison like dependency arrays for hooks or even [arguments passed to a `memize`-d function](https://github.com/google/site-kit-wp/pull/2690) 🙂 

**TL;DR** – since the object is only used to produce a string from the `useSelect`, there should be no problem with the way it is now.